authors/apps/authentication/tests/test_authentication.py

```diff
@@ -1,29 +1,125 @@
 import json
 from django.urls import reverse
+from rest_framework import status
+from ..models import User
 from rest_framework.test import APITestCase, APIClient
 
 
 class UserRegistrationAPIViewTestCase(APITestCase):
     
     def setUp(self):
         self.client = APIClient()
+        self.user_data = {
+                    'user': {
+                        'email': "test@testuser.com",
+                        'username': "testuser",
+                        'password': "password"
+                    }
+                }
 
 
     def test_user_registration(self):
         """
         Test to verify that a post call with user valid data
         """
-        url = reverse("authentication:register")
-
-        user_data = {
-                    'user': {
-                        'email': "test@testuser.com",
-                        'username': "testuser",
-                        'password': "password"
-                    }
-                } 
+        url = reverse("authentication:register") 
 
-        response = self.client.post(url, user_data, format="json")
+        response = self.client.post(url, self.user_data, format="json")
         result = json.loads(response.content)
         self.assertEquals(201, response.status_code)
         self.assertEqual("testuser", result['user']['username'])
+
+
+class UserLoginAPIViewTestCase(APITestCase):
+    """ 
+    Test to test the loginAPIView in the views file.
+    """
+
+    def setUp(self):
+        
+        self.url = reverse("authentication:login")
+        self.test_user = {
```

Comment:
@neelxie I notice a lot duplication of the code under the `setUp()` in every test case class. I am of a view we have a single test case class with a single `setUp()` and have the test methods in this single class.

Reply:
Thanks @alexxsanya  for the response. However, following the single responsibility principle that in simple terms states that `do one thing and do it well`, I would recommend @neelxie  to maintain all the classes when one is inheriting from the other. By this I think we shall be following the [single responsibilty principle](https://itnext.io/solid-principles-explanation-and-examples-715b975dcad4)

Reply:
@alexxsanya I have made changes as you requested thank you.